openmc/statepoint.py

```diff
@@ -524,6 +535,61 @@ def add_volume_information(self, volume_calc):
         if self.summary is not None:
             self.summary.add_volume_information(volume_calc)
 
+    def match_tally(self, tally):
+        """
+        Find a tally with an exact specification match.
+
+            .. versionadded 0.13.4
+
+        Parameters
+        ----------
+        tally : openmc.Tally instance
+            The Tally object to match.
+
+        Returns
+        -------
+        None or openmc.Tally
+        """
+        try:
+            sp_tally = self.get_tally(tally.scores,
+                                      tally.filters,
+                                      tally.nuclides,
+                                      tally.name,
+                                      tally.id,
+                                      tally.estimator,
+                                      exact_filters=True,
+                                      exact_nuclides=True,
+                                      exact_scores=True)
+        except LookupError:
+            sp_tally = None
+
+        # some additional checks
+        if tally.derivative != sp_tally.derivative:
+            sp_tally = None
+
+        if tally.multiply_density != sp_tally.multiply_density:
+            sp_tally = None
+
+        return sp_tally
+
+    def contains_tally(self, tally):
+        """
+        Determine whether or not the statepoint contains an exact match for this tally.
+
+            .. versionadded 0.13.4
```

Comment:
First the `.. versionadded` number seems wrong, and is missing `:`. 

Secondly would `__contains__` be more intuitive?

Reply:
I gave this a try -- and I mostly liked it but for the partial support. The syntax of `tally in statepoint` is definitely enticing, but I think it's a little confusing. I think something like `tally in statepoint.tallies` is a little more clear. Sadly we can't quite get that b/c the `tallies` attribute is a dict but we can get close!

What I've done here is add an `Tally.__eq__` method so we can use:

```python
if tally in statepoint.tallies.values():
    do_the_things(tally)
```
Lemme know what you think!

Reply:
Yeah if you aren't willing to make an object for `statepoint.tallies` I'd use the implementation as is.

Reply:
> Yeah if you aren't willing to make an object for `statepoint.tallies`

Can you elaborate on this?

Reply:
You could do something like:

``` python
class Tallies(dict):
    def __contains__(self, other):
      pass
```

Shameless plug: https://github.com/idaholab/MontePy/blob/develop/montepy/numbered_object_collection.py. It's actually really convenient to have a data collection that you fully control. This is a separate issue that I need to open.

Reply:
See #2953